src/System.Threading/tests/MonitorTests.cs

```diff
@@ -189,35 +190,217 @@ public static void TryEnter_Invalid()
             bool lockTaken = false;
             var obj = new object();
 
-            AssertExtensions.Throws<ArgumentNullException>("obj", () => Monitor.TryEnter(null));
-            AssertExtensions.Throws<ArgumentNullException>("obj", () => Monitor.TryEnter(null, ref lockTaken));
-            AssertExtensions.Throws<ArgumentNullException>("obj", () => Monitor.TryEnter(null, 1));
-            AssertExtensions.Throws<ArgumentNullException>("obj", () => Monitor.TryEnter(null, 1, ref lockTaken));
-            AssertExtensions.Throws<ArgumentNullException>("obj", () => Monitor.TryEnter(null, TimeSpan.Zero));
-            AssertExtensions.Throws<ArgumentNullException>("obj", () => Monitor.TryEnter(null, TimeSpan.Zero, ref lockTaken));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, ref lockTaken));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, 1));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, 1, ref lockTaken));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, TimeSpan.Zero));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, TimeSpan.Zero, ref lockTaken));
 
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("millisecondsTimeout", () => Monitor.TryEnter(null, -1));
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("millisecondsTimeout", () => Monitor.TryEnter(null, -1, ref lockTaken));
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("timeout", () => Monitor.TryEnter(null, TimeSpan.FromMilliseconds(-1)));
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("timeout", () => Monitor.TryEnter(null, TimeSpan.FromMilliseconds(-1), ref lockTaken));
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.TryEnter(obj, -2));
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.TryEnter(obj, -2, ref lockTaken));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("timeout", () => Monitor.TryEnter(obj, TimeSpan.FromMilliseconds(-2)));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("timeout", () => Monitor.TryEnter(obj, TimeSpan.FromMilliseconds(-2), ref lockTaken));
 
             lockTaken = true;
             AssertExtensions.Throws<ArgumentException>("lockTaken", () => Monitor.TryEnter(obj, ref lockTaken));
-            lockTaken = true;
+            Assert.True(lockTaken);
             AssertExtensions.Throws<ArgumentException>("lockTaken", () => Monitor.TryEnter(obj, 0, ref lockTaken));
-            lockTaken = true;
+            Assert.True(lockTaken);
             AssertExtensions.Throws<ArgumentException>("lockTaken", () => Monitor.TryEnter(obj, TimeSpan.Zero, ref lockTaken));
         }
 
+        [Fact]
```

Comment:
Is it reasonable to break this test into several so it's easier to deal with if it fails?

Reply:
I put it in one test since it's testing one related set of behaviors. I could split the non-timeout batch from the timeout batch if that's preferable but it should be fairly easy to tell from the stack trace and line numbers what went wrong. Currently the expected timeout wait duration is 50 ms, so splitting it up per case would take about 100 ms per case, they are tested in parallel currently. I think it's a bit easier to follow in this way.